Including cstring in a clad program throws errors

Minimum reproducible example:
```cpp
#include <iostream>
#include "clad/Differentiator/Differentiator.h" // order of 'Differentiator.h` and `cstring` is important
#include <cstring>                              

int main() {}
```

Result:
```
In file included from bug-cstring-include.cpp:3:
In file included from /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/cstring:42:
/usr/include/string.h:122:14: error: exception specification in declaration does not match previous declaration
extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
             ^
/home/parth/clubs/open-source/clad/pristine/builds/build-10/inst/include/clad/Differentiator/Differentiator.h:23:9: note: previous declaration is here
  char* strcpy (char* destination, const char* source);
        ^
In file included from bug-cstring-include.cpp:3:
In file included from /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/cstring:42:
/usr/include/string.h:385:15: error: exception specification in declaration does not match previous declaration
extern size_t strlen (const char *__s)
              ^
/home/parth/clubs/open-source/clad/pristine/builds/build-10/inst/include/clad/Differentiator/Differentiator.h:24:10: note: previous declaration is here
  size_t strlen(const char*);
         ^
2 errors generated.
/usr/lib/llvm-10/bin/clang++ -g -Xclang -add-plugin -Xclang clad -Xclang -load -Xclang /home/parth/clubs/open-source/clad/pristine/builds/build-10/inst/lib/clad.so -I/home/parth/clubs/open-source/clad/pristine/builds/build-10/inst/include -x c++ -std=c++11 bug-cstring-include.cpp -o build/bug-cstring-include && build/bug-cstring-include
```

--- a/include/clad/Differentiator/Differentiator.h
+++ b/include/clad/Differentiator/Differentiator.h
@@ -20,12 +20,12 @@
 
 extern "C" {
   int printf(const char* fmt, ...);
-  char* strcpy (char* destination, const char* source);
-  size_t strlen(const char*);
 #if defined(__APPLE__) || defined(_MSC_VER)
+  extern size_t strlen(const char*);
   void* malloc(size_t);
   void free(void *ptr);
 #else
+  extern size_t strlen(const char*) __THROW __attribute_pure__ __nonnull ((1));
   void* malloc(size_t) __THROW __attribute_malloc__ __wur;
   void free(void *ptr) __THROW;
 #endif

